### PR TITLE
redirect handled, close #6

### LINKED ISSUE #6: Authentication components and routes
This issues is related to #1 where the plan was to add `authentication` feature (login, register) but for now the task is to rewrite them and add new routes for that.

Note: Some features aren't finished (yet) in our [core platform](https://github.com/doesangueorg/doesangue-core), like **`logout`**, **`password reset`** and **`recovery`**

* **Authentication**
   - [x] `/auth/signin` (Generate a new token for User and log he in)
   - [x] `/auth/signup` (Register a new user and generate the token for **login**)
   - [ ]  `/auth/logout` (Invalidate the token and destroy's `User` session)
   - [ ] `/auth/password/recover` (Ask for a link to reset a password)
   - [ ] `/auth/password/reset` (After the User receives the link in his inbox he check if he can change the **`passwd`**)

One point to **pay attention**: all components related to authentication should be inside `src/components/auth/` organized by folders, like **`components/auth/password/Reset.vue`** and **`components/auth/password/Recover.vue`** to keep this organized..

### PR DESCRIPTION
**Pull request by smkamranqadri**
User will redirect to campaigns page if already loggedin.

if user login then user will redirect to campaigns page.

if user register then user will redirect to campaigns page.

### CODE REVIEW

File: src/components/auth/Login.vue [hunk 1]
```diff
@@ -1,37 +1,46 @@
...
-Vue.use(VueLocalStorage)
+import { mapGetters, mapActions } from 'vuex'
 
 export default {
   data () {
     return {
-      email: '',
-      password: '',
-      failed: false
+      user: {
```
Comment (JoseCage) [inline src/components/auth/Login.vue]: @smkamranqadri I can' understand why we have this test data as a default value.. :confused:

File: src/components/auth/Register.vue [hunk 2]
```diff
@@ -1,45 +1,54 @@
...
-import axios from 'axios'
-import Vue from 'vue'
-import VueLocalStorage from 'vue-localstorage'
-Vue.use(VueLocalStorage)
+import { mapGetters, mapActions } from 'vuex'
 
 export default {
   data () {
     return {
       user: {
```
Comment (JoseCage) [inline src/components/auth/Register.vue]: We have the same here...

File: src/components/auth/Register.vue [hunk 3]
```diff
@@ -1,45 +1,54 @@
...
-        last_name: '',
-        username: '',
-        birthdate: '',
-        password: ''
+        email: 'test@user.com',
+        first_name: 'test',
+        last_name: 'user',
+        username: 'testuser',
+        birthdate: '1987-12-31',
+        password: '12345678'
```
Comment (JoseCage) [inline src/components/auth/Register.vue]: Oops, i'm talking about this fields with *test user* data as default value..

Comment (smkamranqadri) [reply src/components/auth/Register.vue]: So sorry about that.

**Comment by JoseCage** [comment]
@smkamranqadri  Let me just make some tests before the merge..

**Comment by JoseCage** [comment]
I make a simple tests and login works fine. Redirect to `/campaigns` after a successful login but if fails show the error. :+1: :+1: :+1:  
But as i see in `vue-devtolls`, the mutation name is `REGISTER` when the action is `LOGIN`, there is no problem with that?. I'm not a vue (frontend) developer, so sorry..

_Screenhot with my `vue-devtools` data_
screenshot-localhost-8080-2017-11-01-15-17-13-368

**Comment by smkamranqadri** [comment]
Sure, I will check. It should be LOGINSUCCESS.

also, please check register too.

**Comment by JoseCage** [comment]
> also, please check register too.

I have tried yes, but returned the **CORS** error, i will check if it's a configuration fail from the API or it can be done in application itself.

**Comment by JoseCage** [comment]
Hei @smkamranqadri i will merge this to keep the project running.. When you can please fix the mutations name :smile:

**Comment by smkamranqadri** [comment]
Sorry I missed that, sent new pull request.

**Comment by JoseCage** [comment]
> Sorry I missed that, sent new pull request.

Thanks @smkamranqadri i will check..

**Pull request merged**